### PR TITLE
Dedicate fed1 node for s2s tests

### PR DESCRIPTION
**Pull request by arcusfelis**
Separate s2s and global distribution test nodes.
Global distribution tests should use new node reg1 (region 1)
localhost should not be loaded on fed1, because it would make
any additional s2s tests hard to implement.
An example of such test is muc and s2s combination.
Remove debug printing to console from mod_global_distrib_SUITE.
Use generic rpc function provided by escalus from mod_global_distrib_SUITE.
Update docs.

This PR addresses "I can't merge #1311" without this patch.

### CODE REVIEW

**Comment by kzemek** [comment]
> This PR addresses "I can't merge #1311" without this patch.
Can you elaborate?

FWIW, I reused fed1 for global distribution because the pain of testing increases nonlinearly with the number of nodes (esp. for local tests).

**Comment by arcusfelis** [comment]
@kzemek sure, that PR has additional tests for MUC over s2s.
So, the tests would not work, because stanzas, routed to "muc.localhost" would e routed to the local node (i.e. fed1). I need to remove "localhost" from fed1 to fix that tests.

**Comment by arcusfelis** [comment]
But overall, any XMPP interaction should work over s2s.
The fact that we don't test every feature over s2s can hide some bugs.
So, one of the way to start testing, is to override user specs in init_per_group and test several user-configurations.

**Comment by arcusfelis** [comment]
yeah, a bit too many nodes :)
But separate nodes provide better isolation - that means easier debugging/refactoring.

File: rel/reg1.vars.config [hunk 1]
```diff
@@ -0,0 +1,37 @@
+%% This node is for global distribution testing.
+%% reg is short for region.
+%% "localhost" host should be defined.
+{hosts, "[ \"reg1\", \"localhost\" ]"}.
```
Comment (ludwikbukowski) [inline rel/reg1.vars.config]: `"reg1"` domain not needed

Comment (arcusfelis) [reply rel/reg1.vars.config]: done.

Comment (fenek) [reply rel/reg1.vars.config]: It's not funny. Every host in GD setup needs both global domain and local domain.

File: test.disabled/ejabberd_tests/tests/s2s_SUITE.erl [hunk 2]
```diff
@@ -96,13 +96,13 @@ init_per_suite(Config0) ->
 
     Config1 = [{s2s_opts, S2S} | escalus:init_per_suite(Config0)],
     Config2 = [{escalus_user_db, xmpp} | Config1],
-    escalus:create_users(Config2, escalus:get_users([alice2, alice_bis, bob])).
+    escalus:create_users(Config2, escalus:get_users([alice2, alice, bob])).
```
Comment (fenek) [inline test.disabled/ejabberd_tests/tests/s2s_SUITE.erl]: What is the reason behind these changes?

Comment (arcusfelis) [reply test.disabled/ejabberd_tests/tests/s2s_SUITE.erl]: Reverts s2s_SUITE back before 85cc37284fae81f25d14f71cfb3418f4a2402b83.
In that commit, alice was replaced with alice_bis, because it's was not possible to use alice in s2s.
I reverted it back, because we use alice everywhere else.

**Comment by arcusfelis** [comment]
@fenek ok, I've updated "header" of reg1.vars.config, so we would not remove some of the hosts, trying to "optimize" stuff :)
Still not sure, how mim1 host works without a "local" host.

**Comment by fenek** [comment]
@arcusfelis 
`localhost.bis` is the local host of mim1.
